js/ui/searchbar.js

```diff
@@ -0,0 +1,178 @@
+define([
+    "sessions",
+    "command",
+    "editor",
+    "storage/file",
+    "storage/settingsProvider",
+    "ui/statusbar",
+    "ui/projectManager",
+  ], function(sessions, command, editor, File, Settings, status, project) {
+  
+  var Searchbar = function() {
+    var self = this;
+    this.element = document.find(".searchbar");
+    this.input = this.element.find(".search-box");
+    this.maxMatches = 50;
+    Settings.pull("user").then(function(settings) {
```

Comment:
Take a look at the `settings` module that's used plugin-style elsewhere. It lets us force settings to be preloaded before this module runs, so that you can use `Settings.get()` synchronously, and not have to worry about race conditions like this.